gabbi/driver.py

```diff
@@ -261,11 +262,8 @@ def test_suite_from_yaml(loader, test_base_name, test_yaml, test_directory,
     try:
         test_data = test_yaml['tests']
     except KeyError:
-        raise GabbiFormatError(
-            'malformed test file, "tests" key required')
+        raise GabbiFormatError('malformed test file, "tests" key required')
     except TypeError:
-        # Swallow this exception as displaying it does not shine a
```

Comment:
Please don't remove this.

You're welcome to improve it (or the message it is commenting upon), but it is providing a useful piece of information:

* The previous KeyError is caused by the 'tests' key not existing and we express exactly that in the error message
* The TypeError is caused by test_yaml not being a dict (for whatever reason) and I couldn't think of a clean or concise way to say "we expected this but got this" without doing a bit of extra work so I left the comment to excuse the sin of raising a new exception without reference to the exception that inspired it (and that is very much a sin).